pkg/render/logstorage.go

```diff
@@ -525,22 +522,112 @@ func (es elasticsearchComponent) elasticsearchCluster() *esv1.Elasticsearch {
 					},
 				},
 			},
-			NodeSets: []esv1.NodeSet{
-				{
-					Count: int32(nodeConfig.Count),
-					Name: nodeSetName(pvcTemplate),
-					Config: &cmnv1.Config{
-						Data: map[string]interface{}{
-							"node.master": "true",
-							"node.data":   "true",
-							"node.ingest": "true",
+			NodeSets: es.nodeSets(),
+		},
+	}
+}
+
+// nodeSets calculates the number of NodeSets needed for the Elasticsearch cluster. Multiple NodeSets are returned only
+// if the "nodeSets" field has been set in the LogStorage CR. The number of Nodes for the cluster will be distributed as
+// evenly as possible between the NodeSets.
+func (es elasticsearchComponent) nodeSets() []esv1.NodeSet {
+	nodeConfig := es.logStorage.Spec.Nodes
+	pvcTemplate := es.pvcTemplate()
+
+	var nodeSets []esv1.NodeSet
+	if nodeConfig.NodeSets == nil {
+		nodeSet := es.nodeSetTemplate()
+		nodeSet.Name = nodeSetName(pvcTemplate)
+		nodeSet.Count = int32(nodeConfig.Count)
+		nodeSet.PodTemplate = es.podTemplate()
+
+		nodeSets = append(nodeSets, nodeSet)
+	} else {
+		baseNumNodes := nodeConfig.Count / int64(len(nodeConfig.NodeSets))
```

Comment:
What about the case where NodeSets is not nil and Count is zero/unset?
Won't we end up with no nodeSets? Is that valid?

Reply:
Good point! I'll make sure the "if" is hit in that case.

Reply:
I actually think this will panic, can't divide by 0. Rookie mistake.

Reply:
I'm glad you noticed that but I was meaning if `nodeConfig.Count` was 0.

Reply:
If that's 0 in either case there will be no nodes set. We should probably default or warn about this, I'll do this in a separate PR.